orchestrator/utils/websocket.py

```diff
@@ -0,0 +1,80 @@
+import structlog
+import asyncio
+from httpx import AsyncClient
+from broadcaster import Broadcast
+from starlette.concurrency import run_until_first_complete
+from fastapi.websockets import WebSocket
+from fastapi.exceptions import HTTPException
+from orchestrator.security import oidc_user, opa_security_default
+from orchestrator.types import State
+from orchestrator.utils.json import json_dumps
+from orchestrator.forms import generate_form
+
+logger = structlog.get_logger(__name__)
+
+
+class WebsocketManager:
+    def __init__(self, broadcast_url: str):
+        self.broadcast = Broadcast(broadcast_url)
+
+    async def authorize(self, websocket: WebSocket, token: str):
+        try:
+            async with AsyncClient() as client:
+                user = await oidc_user(websocket.url, async_request=client, token=token)
+                await opa_security_default(websocket, user_info=user, async_request=client)
+        except HTTPException as e:
+            return {"error": vars(e)}
+        return
+
+    async def connect(self, websocket, pid):
+        await self.broadcast.connect()
+
+        await run_until_first_complete(
+            (self.sender, {"websocket": websocket, "pid": pid}),
+        )
+
+    async def disconnect(self, websocket, code=1000, reason=""):
```

Comment:
Magic constant 1000?

Reply:
There should already be a constant for this somewhere

Reply:
where can this be found? I can't find constants for it?
I could create constant for it myself.

Reply:
from websockets import ....

Reply:
it doesn't export the constants for it

Reply:
found it in `from fastapi import status`.